Starcounter.Startup/Routing/Middleware/InteractionScopeMiddleware .cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Reflection;
+
+namespace Starcounter.Startup.Routing.Middleware
+{
+    public class InteractionScopeMiddleware : IPageMiddleware
+    {
+        private readonly InteractionScopeAttribute.InteractionScopeMode _defaultValue;
+
+        public InteractionScopeMiddleware(InteractionScopeAttribute.InteractionScopeMode defaultValue = InteractionScopeAttribute.InteractionScopeMode.AttachOrCreate)
+        {
+            _defaultValue = defaultValue;
+        }
+
+        public Response Run(RoutingInfo routingInfo, Func<Response> next)
+        {
+            var attribute = routingInfo.SelectedPageType.GetCustomAttribute<InteractionScopeAttribute>();
+
+            return GetResponseByMode(attribute?.Value ?? _defaultValue, next);
+        }
+
+        private Response GetResponseByMode(InteractionScopeAttribute.InteractionScopeMode mode, Func<Response> next)
+        {
+            switch (mode)
+            {
+                case InteractionScopeAttribute.InteractionScopeMode.AttachOrCreate:
+                    return Db.Scope(next);
+                case InteractionScopeAttribute.InteractionScopeMode.AlwaysCreate:
+                    return new Transaction().Scope(next);
+                case InteractionScopeAttribute.InteractionScopeMode.None:
```

Comment:
I think `None` is kind of misleading. As a developer I would assume it means "ensure that there is no attached interaction scope". However, in practice it rather means "if there is a scope already, then attach to it; otherwise do nothing". I can't see a valid use case for either behavior, so I suggest to drop `None` altogether.